packages/prices/src/prices.ts

```diff
@@ -0,0 +1,179 @@
+   * Find Total Price
+  // Case 1: Base Price per day
+  // Case 2: Base price per kilometer
+  // Case 3: Base price per day + X$ per kilometer over Y range
+  // Case 4: Base price per hour
+  // Case 5: Different price per day
+  // Case 6: Monthly Subscription of X$
+  // Case 7: Discount for Y+ days
+  */
+  async calculateTotalPrice(
+    _priceSpecifications: PriceSpecification[],
+    _rentalCarReservation: RentalCarReservation,
+    _ligoAgreementState: LigoAgreementState
+  ) {
+    let sum = 0;
+    for (let i = 0; i < _priceSpecifications.length; i++) {
+      if (_priceSpecifications[i].referenceQuantity?.unitCode === "DAY") {
+        sum =
+          sum +
+          (await this._totalPriceDays(
+            _priceSpecifications[i],
+            _rentalCarReservation
+          ));
+      } else if (
+        _priceSpecifications[i].referenceQuantity?.unitCode === "KMT" ||
+        _priceSpecifications[i].referenceQuantity?.unitCode === "SMI"
+      ) {
+        sum =
+          sum +
+          (await this._totalPriceKM(
+            _priceSpecifications[i],
+            _ligoAgreementState
+          ));
+      } else if (
+        _priceSpecifications[i].referenceQuantity?.unitCode === "HUR"
+      ) {
+        sum =
+          sum +
+          (await this._totalPriceHour(
+            _priceSpecifications[i],
+            _rentalCarReservation
+          ));
+      } else if (
+        _priceSpecifications[i].referenceQuantity?.unitCode === "MON"
+      ) {
+        sum = sum + (await this._totalMonthlySub(_priceSpecifications[i]));
+      }
+    }
+    return sum;
+  }
+
+  private async _totalPriceDays(
+    _priceSpecification: PriceSpecification,
+    _rentalCarReservation: RentalCarReservation
+  ) {
+    //Filter ValidFrom and ValidThrough
+    const validF = new Date(
+      _priceSpecification.validFrom
+        ? _priceSpecification.validFrom
+        : _rentalCarReservation.pickupTime
+    );
+    const validT = new Date(
+      _priceSpecification.validThrough
+        ? _priceSpecification.validThrough
+        : _rentalCarReservation.dropoffTime
+    );
+    const dropoffT = new Date(_rentalCarReservation.dropoffTime);
+    const pickupT = new Date(_rentalCarReservation.pickupTime);
+    const a = validF >= pickupT ? validF : pickupT;
+    const b = validT <= dropoffT ? validT : dropoffT;
+
+    // For discount maxValue
+    if (_priceSpecification.eligibleQuantity?.maxValue != null) {
+      return (
+        _priceSpecification.eligibleQuantity?.maxValue *
+        _priceSpecification.price
+      );
+    }
+    // For discount minValue
+    if (_priceSpecification.eligibleQuantity?.minValue != null) {
+      return (
+        ((await this._dateDiffInDays(a, b)) -
+          _priceSpecification.eligibleQuantity?.minValue) *
+        _priceSpecification.price
+      );
+    }
+    // Normal days
+    return (await this._dateDiffInDays(a, b)) * _priceSpecification.price;
+  }
+
+  private async _totalPriceKM(
```

Comment:
Is this function assuming that the `unitCode` for the `startOdometer` and `endOdometer` is the same as the `PriceSpecification`? This isn't a safe assumption. What should be done if the units are not the same?

To keep it simple, I think a per distance fee should only be charged if there is a `PriceSpecification` for the correct unit and leave it up to the host to published a per distance fee for each unit separately.